pkg/access_token/access_token_file_test.go

```diff
@@ -0,0 +1,237 @@
+package access_token
+
+import (
+	. "github.com/smartystreets/goconvey/convey"
+	"os"
+	"reflect"
+	"testing"
+)
+
+type ProxyHandlerTokenFile struct {
+	AccessToken AccessTokenHandler
+}
+
+func TestAccessTokenFile(t *testing.T) {
```

Comment:
There seems to be enough overlap between the tests for AccessTokenFile and AccessTokenMemory to warrant splitting out tests for any given `AccessTokenHandler` implementation to
1. Tests that validate the behavior expected of all AccessTokenHandlers
2. Tests that validate the behavior that is specific to this particular implementation e.g. Checking that a file is indeed delete for the AccessTokenFile

That should DRY things up a bit.

Reply:
not sure i understand. what will be the files that you'll have? can you write here some template?

Reply:
Something like below. The idea being to leverage a [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests ) like pattern 
```
type AccessTokenHandler interface {}
type AccessTokenFile struct {}
type AccessTokenMemory struct {}

type AccessTokenHandlerTestCase struct {
	name  string
	handlerConstructor func() AccessTokenHandler
}
var accessTokenHandlerTests = []AccessTokenHandlerTestCase{
	{
		"AccessTokenMemory",
		func() AccessTokenHandler {
			return AccessTokenMemory{}
		},
	},
	{
		"AccessTokenMemory",
		func() AccessTokenHandler {
			return AccessTokenFile{}
		},
	},
}

func TestAccessTokenImplentation(t *testing.T, tc AccessTokenHandlerTestCase) {
	Convey("Read", t, func() {
		accessTokenHandler := tc.handlerConstructor()

			Convey("Given an access token with data saved in memory", func() {
			dataActual := []byte{'t', 'e', 's', 't'}
			_ = accessTokenHandler.Write(dataActual)

			Convey("When running Read method", func() {
				dataExpected, err := accessTokenHandler.Read()

				Convey("Finishes without raising an error", func() {
					So(err, ShouldEqual, nil)
				})

				Convey("Returns the data the was written", func() {
					eq := reflect.DeepEqual(dataActual, dataExpected)
					So(eq, ShouldEqual, true)
				})
			})
		})

	})
}
func TestAccessTokenImplentations(t *testing.T) {
	for _, at := range accessTokenHandlerTests {
		TestAccessTokenImplentation(t, at)
	}
}
```

Reply:
i actually think this decreases readability. the only method that is the same (for now, future implementations may differ) is `Read` and the rest will be in its own file anyway. i think that having each class's test in its own file is better also for readability and for future development

Reply:
I'm starting to agree with you here. Looking at Write and Delete those seem to have very implementation specific. 

My reasoning was that implementations of an interface implicitly have an expected behavior when the methods are called wherever they are called. There would be tests that would validate this. These tests would be run on any new implementation as a smoke test/sanity check - table driven test would allow these to be run in a DRY fashion.

Reply:
i see but we will still need to have the separate tests. for example, we could have a generic test for `Write` in which we will verify that it wrote the content but then again we will need a separate test for `AccessTokenFile` to verify that the data was written to the file specified. This test will probably be very similar if not identical to the generic one so i don't see a benefit here.

@doodlesbykumbi can you write you action item here? do you still think i should add the generic test file?

Reply:
I think it's fine to let it be here.